packages/block-library/src/image/view.js

```diff
@@ -77,269 +77,278 @@ function handleScroll( ctx ) {
-			ctx.hideAnimationEnabled = true;
-			if ( ctx.lightboxEnabled ) {
-				// We want to wait until the close animation is completed
-				// before allowing a user to scroll again. The duration of this
-				// animation is defined in the styles.scss and depends on if the
-				// animation is 'zoom' or 'fade', but in any case we should wait
-				// a few milliseconds longer than the duration, otherwise a user
-				// may scroll too soon and cause the animation to look sloppy.
-				setTimeout( function () {
-					window.removeEventListener( 'scroll', scrollCallback );
-					// If we don't delay before changing the focus,
-					// the focus ring will appear on Firefox before
-					// the image has finished animating, which looks broken.
-					ctx.lightboxTriggerRef.focus( {
-						preventScroll: true,
-					} );
-				}, 450 );
-
-				ctx.lightboxEnabled = false;
-			}
+const { state, actions, callbacks } = store(
+	'core/image',
+	{
+		state: {
+			windowWidth: window.innerWidth,
+			windowHeight: window.innerHeight,
+			get roleAttribute() {
+				const ctx = getContext();
+				return ctx.lightboxEnabled ? 'dialog' : null;
+			},
+			get ariaModal() {
+				const ctx = getContext();
+				return ctx.lightboxEnabled ? 'true' : null;
+			},
+			get dialogLabel() {
+				const ctx = getContext();
+				return ctx.lightboxEnabled ? ctx.dialogLabel : null;
+			},
+			get lightboxObjectFit() {
+				const ctx = getContext();
+				if ( ctx.initialized ) {
+					return 'cover';
+				}
+			},
+			get enlargedImgSrc() {
+				const ctx = getContext();
+				return ctx.initialized
+					? ctx.imageUploadedSrc
+					: 'data:image/gif;base64,R0lGODlhAQABAAD/ACwAAAAAAQABAAACADs=';
+			},
 		},
-		handleKeydown( event ) {
-			const ctx = getContext();
-			if ( ctx.lightboxEnabled ) {
-				if ( event.key === 'Tab' || event.keyCode === 9 ) {
-					// If shift + tab it change the direction
-					if (
-						event.shiftKey &&
-						window.document.activeElement ===
-							ctx.firstFocusableElement
-					) {
-						event.preventDefault();
-						ctx.lastFocusableElement.focus();
-					} else if (
-						! event.shiftKey &&
-						window.document.activeElement ===
-							ctx.lastFocusableElement
-					) {
-						event.preventDefault();
-						ctx.firstFocusableElement.focus();
-					}
+		actions: {
+			showLightbox( event ) {
+				const ctx = getContext();
+				// We can't initialize the lightbox until the reference
+				// image is loaded, otherwise the UX is broken.
+				if ( ! ctx.imageLoaded ) {
+					return;
 				}
+				ctx.initialized = true;
+				ctx.lastFocusedElement = window.document.activeElement;
+				ctx.scrollDelta = 0;
+				ctx.pointerType = event.pointerType;
+
+				ctx.lightboxEnabled = true;
+				setStyles( ctx, ctx.imageRef );
+
+				ctx.scrollTopReset =
+					window.pageYOffset || document.documentElement.scrollTop;
+
+				// In most cases, this value will be 0, but this is included
+				// in case a user has created a page with horizontal scrolling.
```

Comment:
```suggestion
				/*
				 * In most cases, this value will be 0, but this is included
				 * in case a user has created a page with horizontal scrolling.
				 */
```